### ISSUE TITLE
Allow explicit ordering of message headers

### ISSUE THREAD

**Issue by Synchro** [PHPMailer/PHPMailer#121]
[This bug](https://sourceforge.net/p/phpmailer/bugs/343/) is quite reasonable. We should be able to add headers in specific order, perhaps with prepend/append rather than just add.

**Comment by goizueta** [comment]
Is this feature still desirable?

**Comment by Synchro** [comment]
Yes, but be aware it's very difficult to do because of the inconsistent way that headers are handled in PHPMailer. If it was as simple as adding append/prepend methods I've have done it...

**Comment by chornde** [comment]
This may need some refactoring on all occurancies of $mailHeader and $MIMEheader first, does it? Like, adding setters and getters to a private assoc array that can be imploded later. Just a guess ...

**Comment by Synchro** [comment]
It needs to be more involved than a simple assoc array because you can have multiple headers with the same label. My initial thought is to create header and headerCollection classes, and these would be used within the message as a whole (possibly non-MIME), and also within (possibly nested) MIME parts. This would line up with a notional set of MIME leaf and multipart container classes. Being able to support SMTPUTF8 means that all encoding features need to be fairly dynamic. All in all, a big can of worms...
